tabular/src/autogluon/tabular/models/rf/rf_model.py

```diff
@@ -191,6 +195,81 @@ def _predict_proba(self, X, **kwargs):
         y_pred_proba = self.model.predict_proba(X)
         return self._convert_proba_to_unified_form(y_pred_proba)
 
+    def get_oof_pred_proba(self, X, normalize=None, **kwargs):
+        """X should be the same X passed to `.fit`"""
+        y_oof_pred_proba = self._get_oof_pred_proba(X=X, **kwargs)
+        if normalize is None:
+            normalize = self.normalize_pred_probas
+        if normalize:
+            y_oof_pred_proba = normalize_pred_probas(y_oof_pred_proba, self.problem_type)
+        y_oof_pred_proba = y_oof_pred_proba.astype(np.float32)
+        return y_oof_pred_proba
+
+    # FIXME: Unknown if this works with quantile regression
+    def _get_oof_pred_proba(self, X, y, **kwargs):
+        if self._daal:
+            raise AssertionError('DAAL forest backend does not support out-of-bag predictions.')
+        if not self.model.bootstrap:
+            raise ValueError('Forest models must set `bootstrap=True` to compute out-of-fold predictions via out-of-bag predictions.')
+
+        # TODO: This can also be done via setting `oob_score=True` in model params,
+        #  but getting the correct `pred_time_val` that way is not easy, since we can't time the internal call.
+        if (getattr(self.model, "oob_decision_function_", None) is None and getattr(self.model, "oob_prediction_", None) is None) \
+                and callable(getattr(self.model, "_set_oob_score", None)):
+            X = self.preprocess(X)
+
+            if getattr(self.model, "n_classes_", None) is not None:
+                if self.model.n_outputs_ == 1:
+                    self.model.n_classes_ = [self.model.n_classes_]
+            from sklearn.tree._tree import DTYPE, DOUBLE
+            X, y = self.model._validate_data(X, y, multi_output=True, accept_sparse="csc", dtype=DTYPE)
+            if y.ndim == 1:
+                # reshape is necessary to preserve the data contiguity against vs
+                # [:, np.newaxis] that does not.
+                y = np.reshape(y, (-1, 1))
+            if getattr(y, "dtype", None) != DOUBLE or not y.flags.contiguous:
+                y = np.ascontiguousarray(y, dtype=DOUBLE)
+            self.model._set_oob_score(X, y)
+            if getattr(self.model, "n_classes_", None) is not None:
+                if self.model.n_outputs_ == 1:
+                    self.model.n_classes_ = self.model.n_classes_[0]
+
+        if getattr(self.model, "oob_decision_function_", None) is not None:
+            y_oof_pred_proba = self.model.oob_decision_function_
+            self.model.oob_decision_function_ = None  # save memory
+        elif getattr(self.model, "oob_prediction_", None) is not None:
+            y_oof_pred_proba = self.model.oob_prediction_
+            self.model.oob_prediction_ = None  # save memory
+        else:
+            raise AssertionError(f'Model class {type(self.model)} does not support out-of-fold prediction generation.')
```

Comment:
These errors may be more helpful if they describe an alternative for the user.

Reply:
Generally user should never see this unless they are doing very advanced things.